nodejs/lancedb/embedding/embedding_function.ts

```diff
@@ -231,6 +270,40 @@ export abstract class EmbeddingFunction<
     data: T[],
   ): Promise<number[][] | Float32Array[] | Float64Array[]>;
 
+  /**
+   * Compute source embeddings with retry and rate limiting.
+   * 
+   * @param data The data to create embeddings for
+   * @returns The embeddings
+   */
+  async computeSourceEmbeddingsWithRetry(
+    data: T[],
+  ): Promise<number[][] | Float32Array[] | Float64Array[]> {
+    let embedFunc = () => this.computeSourceEmbeddings(data);
+    
+    // Apply rate limiting if configured
+    if (this.rateLimiterOptions) {
+      const limiter = new RateLimiter(
+        this.rateLimiterOptions.maxCalls,
+        this.rateLimiterOptions.period
+      );
+      embedFunc = limiter.wrap(embedFunc);
+    }
```

Comment:
As written, this will create a new `RateLimiter` for each call to `computeSourceEmbeddingsWithRetry`. Have you considered instead attaching a global `RateLimiter` in the embedding function itself? Then multiple concurrent calls to this method would share the same rate limiter.